database/ffldb/db.go

```diff
@@ -1172,6 +1187,12 @@ func (tx *transaction) StoreBlock(block *btcutil.Block) error {
 		return makeDbErr(database.ErrDriverSpecific, str, err)
 	}
 
+	if block.Height() == btcutil.BlockHeightUnknown {
```

Comment:
I believe this is going to be a problem since the block is inserted before its contextual position is known.  See [here](https://github.com/btcsuite/btcd/blob/3cb87afa2fc078efc2b12d16458cce15daab311c/blockchain/accept.go#L45-L59).

I understand that once everything is fully headers first, versus the partial headers-first approach that is used now, this probably won't be an issue since the block's position will be known by virtue of its header already having been processed.

Reply:
Correct me if I'm wrong, but from what I can tell `maybeAcceptBlock` is never called with an orphan block. The call to `maybeAcceptBlock` is skipped [here](https://github.com/btcsuite/btcd/blob/3cb87afa2fc078efc2b12d16458cce15daab311c/blockchain/process.go#L225) if the previous block is not known. So by the time the block is stored, the height is definitely known I believe.

Reply:
You're right.  Sorry, I've been looking at like 4 different code bases that are all really similar, so I got this one confused with another.  I should've looked before commenting.  It does indeed set the block height about six lines above the code I linked.  Doh!